app/controllers/successes_controller/google_map_coordinates.rb

```diff
@@ -1,55 +1,35 @@
 class SuccessesController
-	# class to retrieve and prepare coordinates for Google Map API, used by successes controller
-	class GoogleMapCoordinates
-		attr_reader :map_collection, :raw_collection
+  # class to retrieve and prepare coordinates for Google Map API, used by successes controller
+  class GoogleMapCoordinates
+    attr_accessor :location, :map_data
 
-		def initialize(raw_collection=[])
-			@raw_collection = raw_collection
-			@duplicate_locations = []
-			@map_collection = []
-		end
+    def initialize(map_data = [])
+      @map_data = map_data
+    end
 
-		def create_raw_collection
-			Adoption.all.each do |adoption|
+    def genarate_map_data
+      generate_map_data_from_adoptions
+    end
 
-				lat = adoption.adopter_account.adopter_profile.location.latitude
-				lon = adoption.adopter_account.adopter_profile.location.longitude
-	
-				unless lat.nil? || lon.nil?
-					@raw_collection << {latitude: lat, longitude: lon}
-				end
-			end
-			@raw_collection
-		end
+    def generate_map_data_from_adoptions
```

Comment:
I think this is creating a N+1 case. We are loading the `Adoption` collection, and then iterating over it and calling the `adopter_account.adopter_profile.location` and `dog.xxx` each iteration. I think we can eager load this collection first using `includes` to get all that data in one query and then iterate over it. This will reduce the number of database calls. 

I think something like `Adoption.includes(adopter_account: { adopter_profile: :location }, dog: [:name, :breed]).all.each do |adoption|` will do it?

Thoughts on that?

Reply:
Nice catch. You are right! Your suggestion would most likely work. I will handle this.